incubator/group/msg.go

```diff
@@ -0,0 +1,59 @@
+package group
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/pkg/errors"
+)
+
+var _ sdk.Msg = &MsgCreateGroup{}
+
+const (
+	msgTypeGroup = "create_group"
+)
+
+func (m MsgCreateGroup) Route() string { return ModuleName }
+func (m MsgCreateGroup) Type() string  { return msgTypeGroup }
+
+// GetSigners returns the addresses that must sign over msg.GetSignBytes()
+func (m MsgCreateGroup) GetSigners() []sdk.AccAddress {
+	// TODO: @aaronc ok with this constraint? We can enforce the signature on creation. Also see `MsgUpdateGroupAdmin.GetSigners()`.
+	return []sdk.AccAddress{m.Admin}
+}
+
+// GetSignBytes returns the bytes for the message signer to sign on
+func (m MsgCreateGroup) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(m))
```

Comment:
Not sure what to do here. Not really intending to support amino, but maybe we should support it just in case...

Reply:
I do understand the concerns about Amino but as long as cosmos-sdk requires it let's not make it our problem. I would suggest we implement all the group functionality and then add a separate task to drop Amino support. Otherwise I may loose time working around problems for integration tests rather than getting the logic done.

Reply:
Thinking a bit further: can we drop JSON here completely and use protobuf instead? This would require the clients to switch to protobuf for signing as well.

Reply:
Let's actually keep as is until Amino tx support actually gets dropped.